src/homework_collectionsassignments_maps/HomeworkMaps.java

```diff
@@ -0,0 +1,54 @@
+package homework_collectionsassignments_maps;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class HomeworkMaps {
+
+    /*1.Дана Map<String, String> map, написать функцию, которая вернет Мап, такую,
+    если в есходной map есть ключ ‘a’ и ‘b’, то нужно создать новый ключ ‘ab’ ссуммой значений от ключей a и b.
+    Примеры:
+    mapAB({"a": "Hi", "b": "There"})  → {"a": "Hi", "ab": "HiThere", "b": "There"}
+    mapAB({"a": "Hi"}) → {"a": "Hi"}mapAB({"b": "There"}) → {"b": "There"}
+
+    */
+
+    public Map<String, String> ab(Map<String, String> input) {
+        Map<String, String> mapAB = new HashMap();
+        if (input.containsKey("a")) mapAB.put("a", input.get("a"));
+        if (input.containsKey("b")) mapAB.put("b", input.get("b"));
```

Comment:
I am not sure this is correct. If a key with a value is already there, makes no sense to put it again with the same value.

Reply:
But they are for different maps -the idea is if there is only key a or only key b in initial map, then key a or key b are nonetheless created in mapAB. Judging by the task examples it is so:
mapAB({"a": "Hi"}) → {"a": "Hi"};  mapAB({"b": "There"}) → {"b": "There"} If the new value is only created when a and b are present in initial map, then of course  these operations are redundant.